libminifi/src/Site2SiteClientProtocol.cpp

```diff
@@ -319,6 +344,64 @@ void Site2SiteClientProtocol::tearDown() {
   _peerState = IDLE;
 }
 
+bool Site2SiteClientProtocol::getPeerList(std::vector<minifi::Site2SitePeerStatus> &peer) {
+  if (establish() && handShake()) {
+    int status = this->writeRequestType(REQUEST_PEER_LIST);
+
+    if (status <= 0) {
+      tearDown();
+      return false;
+    }
+
+    uint32_t number;
+    status = peer_->read(number);
+
+    if (status <= 0) {
+      tearDown();
+      return false;
+    }
+
+    for (int i = 0; i < number; i++) {
```

Comment:
it is config by the NiFI cluster server to let client evenly distribute the node cross peers.
MiNiFI java is doing the similar thing.